### PR TITLE
Fix/minimize mock_time_env.h dependencies

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: (a) own copy of kMicrosInSecond
(b) out-of-line sync point code

Test Plan: FB internal

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -731,6 +731,7 @@ set(SOURCES
         table/table_factory.cc
         table/table_properties.cc
         table/two_level_iterator.cc
+        test_util/mock_time_env.cc
```
Comment (mrambacher) [inline CMakeLists.txt:734]: Shouldn't this be in TEST_LIB_SOURCES and not LIB_SOURCES?

File: src.mk [hunk 2]
```diff
@@ -178,6 +178,7 @@ LIB_SOURCES =                                                   \
   table/table_factory.cc                                        \
   table/table_properties.cc                                     \
   table/two_level_iterator.cc                                   \
+  test_util/mock_time_env.cc                                    \
```
Comment (mrambacher) [inline src.mk:181]: TEST_LIB_SOURCES?

File: test_util/mock_time_env.h [hunk 3]
```diff
@@ -62,29 +64,11 @@ class MockTimeEnv : public EnvWrapper {
   // TODO: this is a workaround for the different behavior on different platform
   // for timedwait timeout. Ideally timedwait API should be moved to env.
   // details: PR #7101.
-  void InstallTimedWaitFixCallback() {
```
Comment (mrambacher) [inline test_util/mock_time_env.h:65]: Should the methods that use std::numeric_limits also be moved or should <limits> be included?

**Review by mrambacher — APPROVED** [review]
LGTM, with the exception of LIB_SOURCES

**Review by jay-zhuang — APPROVED** [review]

**Review by jay-zhuang — APPROVED** [review]
Thanks for the fix.

File: test_util/mock_time_env.h (hunk 3 above)
Comment (pdillinger) [reply test_util/mock_time_env.h:65]: env.h -> limits is sufficient

File: CMakeLists.txt (hunk 1 above)
Comment (pdillinger) [reply CMakeLists.txt:734]: I was just copying from sync_point.cc. I'll try it and see if I have difficulty working out the public as well as internal dependencies.

Comment (jay-zhuang) [reply CMakeLists.txt:734]: Maybe `TESTUTIL_SOURCE` for cmake.

**Pull request closed without merging**